### PR TITLE
[LEV-2512] wallet rescan improvements

### PR DESCRIPTION
**Pull request by sergey-chernikov**

### CODE REVIEW

File: BlocksettleNetworkingLib/Adapters/WalletsAdapter.cpp [hunk 1]
```diff
@@ -888,14 +978,66 @@ void WalletsAdapter::sendBalanceRequest(const std::string &walletId)
...
+   sendWalletChanged(hdWalletId);
+}
+
+void WalletsAdapter::scan(const std::shared_ptr<bs::sync::hd::Wallet>& hdWallet
+   , const std::shared_ptr<bs::sync::Wallet>& wallet
+   , const std::function<void(bs::sync::SyncState)>& cb)
+{
+   logger_->debug("[WalletsAdapter::scan] {}", wallet->walletId());
+   wallets_[wallet->walletId()] = wallet;
+   scanWallet(wallet, true);
```
Comment (goatpig) [inline BlocksettleNetworkingLib/Adapters/WalletsAdapter.cpp:1039]: why is this called twice?

Comment (sergey-chernikov) [reply BlocksettleNetworkingLib/Adapters/WalletsAdapter.cpp:1039]: external and internal addresses (2nd arg)

Comment (goatpig) [reply BlocksettleNetworkingLib/Adapters/WalletsAdapter.cpp:1039]: A scan operation is several minutes long at least. It also cannot happen in parallel. Queuing multiple set of addresses to be rescanned independently is not an efficient use of armorydb. You should batch the rescan of all concerned wallets in a single call.

**Review by goatpig — APPROVED** [review]
utACK

**Pull request merged**